clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/ResizeAsgAtomicOperationConverter.groovy

```diff
@@ -38,4 +40,36 @@ class ResizeAsgAtomicOperationConverter extends AbstractAtomicOperationsCredenti
     converted.credentials = getCredentialsObject(input.credentials as String)
     converted
   }
+
+  /**
+   * Converts legacy resize asg descriptions to their modern equivalent.
+   *
+   * Expects to be run after {@see AsgNameToServerGroupNameDescriptionPreProcessor}.
+   */
+  @Order(0)
+  @Component
+  static class ResizeAsgDescriptionPreProcessor implements AtomicOperationDescriptionPreProcessor {
+    @Override
+    boolean supports(Class descriptionClass, Map description) {
+      return descriptionClass == ResizeAsgDescription
+    }
+
+    @Override
+    Map process(Map description) {
+      if (description.asgs) {
+        return description
+      }
+
+      description.asgs = description.regions.collect {
```

Comment:
This conversion lets us simplify the actual ResizeAsgOperation and not worry about the various edge cases around different possible inputs.